pkg/vmconfig/crypto/crypto_reconciler_pre.go

```diff
@@ -0,0 +1,690 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package crypto
+
+import (
+	"context"
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/vmware/govmomi/crypto"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	byokv1 "github.com/vmware-tanzu/vm-operator/external/byok/api/v1alpha1"
+	kubeutil "github.com/vmware-tanzu/vm-operator/pkg/util/kube"
+	"github.com/vmware-tanzu/vm-operator/pkg/util/paused"
+	"github.com/vmware-tanzu/vm-operator/pkg/vmconfig/crypto/internal"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha3"
+)
+
+type ptrCfgSpec = *vimtypes.VirtualMachineConfigSpec
```

Comment:
Hmm.  Is this a pattern?  What does this buy us compared to just using `*vimtypes.VirtualMachineConfigSpec`?

Reply:
It is used elsewhere in our codebase, yes. The types are exactly the same, and it enables more concise code. Even intellisense shows what it actually is.